### ISSUE TITLE
Add linting to Git hooks

### ISSUE THREAD

**Issue by artem-zakharchenko** [apiaryio/dredd#1321]
I propose to add various linting tools we use as a part of Git workflow.

## Why:

- To make daily contributions more seamless, prevent failing CI on old commits due to linting now being run manually
- To automate linting, because it's meant to be automated
- To ease the first contributor experience and prevent pushing invalid code

## How:

1. Use `husky` for `pre-commit` hook to perform `eslint`.
  1. Alternatively, use `lint-staged` to eslint and prettify code.
1. Use `husky` for `pre-push` hook to use `commitlint` to validate commits.

**Comment by honzajavorek** [comment]
I agree the CI should fail fast, or somehow signalize that the lint is failing fast, otherwise one waits 15 minutes until realizing a commit needs to be fixed or a semicolon is missing.

If the hooks are shared using husky, I'd be cautious about `pre-commit` with `eslint` as that could break my workflow when prototyping or committing work in progress often, but I'd be fine with having it in `pre-push` for sure.

Currently that can be achieved locally using a following workaround:

```bash
$ echo "npm run lint" > .git/hooks/pre-push
$ chmod +x .git/hooks/pre-push
```

I'll try to use that for a start before we make permanent change and have husky as an additional dependency.

**Comment by honzajavorek** [comment]
@artem-zakharchenko Can we close this now? Is everything in place?
